src/main/java/org/testng/internal/PackageUtils.java

```diff
@@ -83,68 +84,92 @@ public static void addClassLoader(final ClassLoader loader) {
       }
     }
 
+    ParamContainer container = new ParamContainer(included,excluded,vResult);
```

Comment:
Sometimes, we should ignore sonar or dig deeper. ;)
Here, the fix is not having a container to mute sonar but try to understand why we need so many params and try to find another design to reach the objectives.